### DISCUSSION TITLE
Implement mmap

### DISCUSSION THREAD

**Discussion by JasonShin** [josdejong/mathjs#2759] [Ideas]
Hi, this is an idea but I thought it would be nice if Math.js can provide a way to deal with large arrays or matrix. If someone is working on a deep learning or machine learning problem, they can easily run into an out of memory issue while dealing with images or a big corpus dataset. I propose implementing https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.memmap.html, which will provide a quick workaround to this issue. 

**Ideation:**
- My initial thought is that this method will only be usable in the Node environment as it requires R/W to the file system.

**Related issues:**
- #874

**Comment by josdejong** [comment]
Thanks for your suggestion Jason. Do you see this as a new type of Matrix, a BinaryMatrix or something like that? Do you already have ideas how would this work concretely in mathjs?

Regarding memory and matrices: we already do have Sparse matrices though they are not suitable for images for example. Just mention it here to keep the discussion complete.

**Comment by JasonShin** [comment]
Thanks for the reply @josdejong. I will do more research first and get back to you with a better planning around this idea. In the meantime, I welcome any feedback or thoughts from others.

**Comment by josdejong** [comment]
Thanks! Feel free to share and discuss any progress and ideas here :)
